src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs

```diff
@@ -98,15 +99,19 @@ public async ValueTask<object> GetAssetAsync(Checksum checksum, CancellationToke
                 Contract.ThrowIfTrue(checksum == Checksum.Null);
 
                 using var checksumPool = Creator.CreateChecksumSet(checksum);
-                using var _ = Creator.CreateResultMap(out var resultPool);
 
-                await scope.FindAssetsAsync(AssetPath.FullLookupForTesting, checksumPool.Object, resultPool, cancellationToken).ConfigureAwait(false);
-                Contract.ThrowIfTrue(resultPool.Count != 1);
+                object? asset = null;
+                await scope.FindAssetsAsync(AssetPath.FullLookupForTesting, checksumPool.Object, (foundChecksum, foundAsset, _) =>
+                {
+                    Contract.ThrowIfTrue(asset != null); // We should only find one asset
+                    Contract.ThrowIfTrue(checksum != foundChecksum);
+                    asset = foundAsset;
+                    return ValueTaskFactory.CompletedTask;
+                }, cancellationToken).ConfigureAwait(false);
```

Comment:
Note; this is a test-accessor.  No concern abotu captures and perf and whatnot here.